src/transform/src/normalize_lets.rs

```diff
@@ -438,6 +438,28 @@ mod inlining {
 
     use mz_expr::{Id, LocalId, MirRelationExpr};
 
+    pub(super) fn inline_lets(
+        expr: &mut MirRelationExpr,
+        inline_mfp: bool,
+    ) -> Result<(), crate::TransformError> {
+        let mut worklist = vec![&mut *expr];
+        while let Some(expr) = worklist.pop() {
+            inline_lets_core(expr, inline_mfp)?;
+            // We descend only into `LetRec` nodes, because they have been already normalized to
```

Comment:
nit: I think the comment is can be slightly improved:

> We descend into top-level `LetRec` nodes (which will exist if inline_lets_core has done some work)
> in order to normalize recursive `LetRec` nodes.

Reply:
> which will exist if inline_lets_core has done some work

Actually, there might be a `LetRec` node here even if `inline_lets_core` didn't do any work.

I edited the comment a bit. So, why it is enough to descend only into `LetRec` nodes, is because `promote_let_rec` ensured that if there are any `LetRec` nodes in a subtree, then one of those has to be at the root. In other words, if there are multiple `LetRec` nodes in a tree, then you can get to all of them by just descending into `LetRec` nodes. 

This is the new comment:
```
            // We descend only into `LetRec` nodes, because `promote_let_rec` ensured that all
            // `LetRec` nodes are clustered near the root. This means that we can get to all the
            // `LetRec` nodes by just descending into `LetRec` nodes, as there can't be any other
            // nodes between them.
```